test/integration/factory.js

```diff
@@ -51,4 +56,56 @@ describe('factory', function() {
             return done();
         });
     });
+
+    it('invalid filters in createCacheTable', function(done) {
+        const TEST_SOURCE_TYPE = 'test-source';
+        const TestSource = Node.create(TEST_SOURCE_TYPE, {
+            table: Node.PARAM.STRING()
+        }, { cache: true });
+        TestSource.prototype.sql = function() {
```

Comment:
Long time I worked on this library, so I might forget a lot of context. So my following question might not make sense: why do we need a `test-source` type?

Reply:
Good question, I'm not sure either, I picked it from the existing test in that file, it's a cached source, it worked for me so I used it (but I guess I could use an actual Source).